src/main/java/tds/exam/results/mappers/OpportunityMapper.java

```diff
@@ -55,75 +69,82 @@
         opportunity.setTaName(session.getProctorName());
         opportunity.setSessionId(session.getId().toString());
         opportunity.setWindowId(exam.getAssessmentWindowId());
-//        opportunity.setWindowOpportunity(); TODO: Add windowOpportunity to ExpandableExam
-        opportunity.setAdministrationCondition(
-            exam.getStatus().getCode().equalsIgnoreCase(ExamStatusCode.STATUS_INVALIDATED)
-                ? ADMINISTRATION_CONDITION_INVALIDATED
-                : "");
+        opportunity.setWindowOpportunity(String.valueOf(expandableExam.getWindowAttempts()));
+        opportunity.setAdministrationCondition(exam.getStatus().getCode().equalsIgnoreCase(ExamStatusCode.STATUS_INVALIDATED)
+            ? ADMINISTRATION_CONDITION_INVALIDATED
+            :  StringUtils.EMPTY);
         opportunity.setDateForceCompleted(expandableExam.getForceCompletedAt() != null
-            ? expandableExam.getForceCompletedAt().toString()
-            : null); //TODO: Verify date format
+            ? expandableExam.getForceCompletedAt().toDateTime().toString()
+            : null);
         opportunity.setClientName(exam.getClientName());
         opportunity.setAssessmentParticipantSessionPlatformUserAgent(exam.getBrowserUserAgent());
-//        opportunity.setEffectiveDate(); TODO: Add effectiveDate to ExpandableExam
+        opportunity.setEffectiveDate(!assessmentWindows.isEmpty()
+            ? assessmentWindows.get(0).getStartTime().toString() //TODO: Verify date format
```

Comment:
I know it seems sort of bizarre that we are just grabbing the first element, regardless of the number of active assessment windows, but this is essentially what they do in the legacy implementation as well (see `ReportingDLL.getEffectiveDate()`. Typically there won't be more than 1 active assessment window for an assessment.